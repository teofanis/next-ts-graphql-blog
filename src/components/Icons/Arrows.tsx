import React from 'react'

interface ArrowsProps {
  direction: 'left' | 'right'
}

const Arrows = ({ direction }: ArrowsProps) => {
  const leftArrowSVG = (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="h-6 w-6 text-white w-full"
      fill="none"
      viewBox="0 0 24 24"
      stroke="currentColor"
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        d="M10 19l-7-7m0 0l7-7m-7 7h18"
      />
    </svg>
  )

  const rightArrowSVG = (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      className="h-6 w-6 text-white w-full"
      fill="none"
      viewBox="0 0 24 24"
      stroke="currentColor"
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        d="M14 5l7 7m0 0l-7 7m7-7H3"
      />
    </svg>
  )
  if (direction === 'left') {
    return leftArrowSVG
  } else if (direction === 'right') {
    return rightArrowSVG
  }

  return null
}

export default Arrows
