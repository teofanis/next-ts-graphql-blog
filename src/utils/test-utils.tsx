import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
// Add in any providers here if necessary:

const Providers: React.FC = ({ children }) => {
  return <>{children}</>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
// override render method
export { customRender as render }
