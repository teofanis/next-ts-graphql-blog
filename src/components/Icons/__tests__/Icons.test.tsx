import Arrows from '@icons/Arrows'
import CalendarIcon from '@icons/CalendarIcon'
import Loader from '@icons/Loader'
import { render, screen, within } from '@utils/test-utils'
import React from 'react'
import { direction } from './__fixtures__/arrows.json'

describe('Icons Tests', () => {
  describe('Calendar Icon Tests', () => {
    it('renders correctly', () => {
      const { container } = render(<CalendarIcon />)
      expect(container).toMatchSnapshot()
    })

    it('should render a calendar SVG icon', () => {
      render(<CalendarIcon />)
      const icon = screen.getByTestId('calendar-icon')
      expect(icon).toBeInTheDocument()
      expect(icon).toBeVisible()
      expect(icon).toHaveAttribute('data-testid', 'calendar-icon')
    })
  })

  describe('Loader Icon Tests', () => {
    it('renders correctly', () => {
      const { container } = render(<Loader />)
      expect(container).toMatchSnapshot()
    })
    it('should render Loader icon', () => {
      const { getByTestId } = render(<Loader />)
      const wrapper = getByTestId('loader-svg')
      expect(wrapper).toBeInTheDocument()
      expect(wrapper.firstChild).toBeTruthy()
      const button = within(wrapper).getByRole('button')
      expect(button).toBeInstanceOf(HTMLButtonElement)
      expect(button).toBeDisabled()
      expect(within(button).getByText('Loading')).toBeInTheDocument()
    })
  })

  describe('Arrow Icon tests', () => {
    it('renders correctly', () => {
      // @ts-ignore
      const { container } = render(<Arrows direction={direction} />)
      expect(container).toMatchSnapshot()
    })
    it('renders a left arrow icon', () => {
      const { getByTestId } = render(<Arrows direction="left" />)
      const arrow = getByTestId('left-arrow-svg')
      expect(arrow).toBeInTheDocument()
      expect(arrow).toBeVisible()
    })

    it('renders a right arrow icon', () => {
      const { getByTestId } = render(<Arrows direction="right" />)
      const arrow = getByTestId('right-arrow-svg')
      expect(arrow).toBeInTheDocument()
      expect(arrow).toBeVisible()
    })

    it('does not render with invalid direction', () => {
      // @ts-expect-error
      render(<Arrows direction="unknown" />)
      const leftArrow = screen.queryByTestId('left-arrow-svg')
      const rightArrow = screen.queryByTestId('right-arrow-svg')
      expect(leftArrow).not.toBeInTheDocument()
      expect(rightArrow).not.toBeInTheDocument()
    })
  })
})
