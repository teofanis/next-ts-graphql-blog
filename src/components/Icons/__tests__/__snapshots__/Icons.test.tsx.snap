// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons Tests Arrow Icon tests renders correctly 1`] = `
<div>
  <svg
    class="h-6 w-6 text-white w-full"
    data-testid="left-arrow-svg"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 19l-7-7m0 0l7-7m-7 7h18"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons Tests Calendar Icon Tests renders correctly 1`] = `
<div>
  <svg
    class="h-6 w-6 inline mr-2 text-pink-500 dark:text-yellow-500"
    data-testid="calendar-icon"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons Tests Loader Icon Tests renders correctly 1`] = `
<div>
  <div
    class="text-center"
    data-testid="loader-svg"
  >
    <button
      class="inline-flex items-center px-4 py-2 border border-transparent text-base leading-6 font-medium rounded-md text-white bg-rose-600 hover:bg-rose-500 focus:border-rose-700 active:bg-rose-700 transition ease-in-out duration-150 cursor-not-allowed"
      disabled=""
      type="button"
    >
      <svg
        class="animate-spin -ml-1 mr-3 h-5 w-5 text-white"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="opacity-25"
          cx="12"
          cy="12"
          r="10"
          stroke="currentColor"
          stroke-width="4"
        />
        <path
          class="opacity-75"
          d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
          fill="currentColor"
        />
      </svg>
      Loading
    </button>
  </div>
</div>
`;
